From a50b9f5e62e26d311f18d3bdab19a9008b8f101c Mon Sep 17 00:00:00 2001
From: Tingguo Cheng <quic_tingguoc@quicinc.com>
Date: Fri, 26 Jul 2024 17:03:29 +0800
Subject: [PATCH 19/30] dt-bindings: power: qcom,rpmpd: document sa7255p

Add compatible for sa7255p platform and add it's defines in include
file.

Signed-off-by: Shazad Hussain <quic_shazhuss@quicinc.com>
Signed-off-by: Tingguo Cheng <quic_tingguoc@quicinc.com>
---
 .../devicetree/bindings/power/qcom,rpmpd.yaml |  1 +
 include/dt-bindings/power/qcom-rpmpd.h        | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/Documentation/devicetree/bindings/power/qcom,rpmpd.yaml b/Documentation/devicetree/bindings/power/qcom,rpmpd.yaml
index 929b7ef9c1bc..c166fda54f42 100644
--- a/Documentation/devicetree/bindings/power/qcom,rpmpd.yaml
+++ b/Documentation/devicetree/bindings/power/qcom,rpmpd.yaml
@@ -34,6 +34,7 @@ properties:
           - qcom,qcs404-rpmpd
           - qcom,qdu1000-rpmhpd
           - qcom,qm215-rpmpd
+          - qcom,sa7255p-rpmhpd
           - qcom,sa8155p-rpmhpd
           - qcom,sa8540p-rpmhpd
           - qcom,sa8775p-rpmhpd
diff --git a/include/dt-bindings/power/qcom-rpmpd.h b/include/dt-bindings/power/qcom-rpmpd.h
index 608087fb9a3d..a4487880b130 100644
--- a/include/dt-bindings/power/qcom-rpmpd.h
+++ b/include/dt-bindings/power/qcom-rpmpd.h
@@ -4,6 +4,25 @@
 #ifndef _DT_BINDINGS_POWER_QCOM_RPMPD_H
 #define _DT_BINDINGS_POWER_QCOM_RPMPD_H
 
+/* SA7255 Power Domain Indexes */
+#define SA7255P_CX	0
+#define SA7255P_CX_AO	1
+#define SA7255P_DDR	2
+#define SA7255P_EBI	3
+#define SA7255P_GFX	4
+#define SA7255P_LCX	5
+#define SA7255P_LMX	6
+#define SA7255P_MMCX	7
+#define SA7255P_MMCX_AO	8
+#define SA7255P_MSS	9
+#define SA7255P_MX	10
+#define SA7255P_MX_AO	11
+#define SA7255P_MXC	12
+#define SA7255P_MXC_AO	13
+#define SA7255P_NSP0	14
+#define SA7255P_NSP1	15
+#define SA7255P_XO	16
+
 /* SA8775P Power Domain Indexes */
 #define SA8775P_CX	0
 #define SA8775P_CX_AO	1
-- 
2.34.1

